integration/test/Testlib/MockIntegrationService.hs

```diff
@@ -107,30 +109,43 @@ withMockServer mkApp go = withFreePortAnyAddr $ \(sPort, sock) -> do
   let tlss = Warp.tlsSettingsMemory (cs mockServerCert) (cs mockServerPrivKey)
   let defs = Warp.defaultSettings {Warp.settingsPort = sPort, Warp.settingsBeforeMainLoop = putMVar serverStarted ()}
   buf <- newChan
-  srv <- async . liftIO . Warp.runTLSSocket tlss defs sock $ mkApp buf
+  srv <- async $ withRunInIO \inIO -> do
+    Warp.runTLSSocket tlss defs sock \req respond -> do
+      inIO $ mkApp buf req (liftIO . respond)
   srvMVar <- UnliftIO.Timeout.timeout 5_000_000 (takeMVar serverStarted)
   case srvMVar of
     Just () -> go sPort buf `finally` cancel srv
     Nothing -> error . show =<< poll srv
 
+lhMockApp :: Chan (Wai.Request, LBS.ByteString) -> LiftedApplication
+lhMockApp = lhMockApp' Nothing
+
 -- | LegalHold service.  Just fake the API, do not maintain any internal state.
-lhMockApp :: Chan (Wai.Request, LBS.ByteString) -> Wai.Application
-lhMockApp ch req cont = do
+lhMockApp' :: Maybe (Value, [Value]) -> Chan (Wai.Request, LBS.ByteString) -> LiftedApplication
+lhMockApp' mks ch req cont = withRunInIO \inIO -> do
   reqBody <- Wai.strictRequestBody req
   writeChan ch (req, reqBody)
-  case (cs <$> pathInfo req, cs $ requestMethod req, cs @_ @String <$> getRequestHeader "Authorization" req) of
-    (["legalhold", "status"], "GET", _) -> cont respondOk
-    (_, _, Nothing) -> cont missingAuth
-    (["legalhold", "initiate"], "POST", Just _) -> cont initiateResp
-    (["legalhold", "confirm"], "POST", Just _) -> cont respondOk
-    (["legalhold", "remove"], "POST", Just _) -> cont respondOk
-    _ -> cont respondBad
+  inIO do
+    (nextLastPrekey, threePrekeys) <-
+      case mks of
+        Nothing ->
+          (,)
+            <$> getLastPrekey
+            <*> replicateM 3 getPrekey
```

Comment:
use the ones provided in the `Maybe` and if they're `Nothing`, use the ones from the `IORefs` in the `Reader`